test/time_test.py

```diff
@@ -137,86 +176,67 @@ def test_tick(self):
-        pygame.time.wait(10)  # incur delay between ticks that's faster than sample_fps
-        self.assertGreaterEqual(c.tick_busy_loop(sample_fps), (second_length/sample_fps) - shortfall_tolerance)
-        pygame.time.wait(200)  # incur delay between ticks that's slower than sample_fps
-        self.assertGreaterEqual(c.tick_busy_loop(sample_fps), (second_length/sample_fps) - shortfall_tolerance)
-
-        high_fps = 500
-        self.assertGreaterEqual(c.tick_busy_loop(high_fps), (second_length/high_fps) - shortfall_tolerance)
-
-        low_fps = 1
-        self.assertGreaterEqual(c.tick_busy_loop(low_fps), (second_length/low_fps) - shortfall_tolerance)
-
-        low_non_factor_fps = 35  # 1000/35 makes 28.5714285714
-        frame_length_without_decimal_places = int(second_length/low_non_factor_fps)  # Same result as math.floor
-        self.assertGreaterEqual(c.tick_busy_loop(low_non_factor_fps), frame_length_without_decimal_places - shortfall_tolerance)
-
-        high_non_factor_fps = 750  # 1000/750 makes 1.3333...
-        frame_length_without_decimal_places_2 = int(second_length/high_non_factor_fps)  # Same result as math.floor
-        self.assertGreaterEqual(c.tick_busy_loop(high_non_factor_fps), frame_length_without_decimal_places_2 - shortfall_tolerance)
-
+        if not _is_almost_equal(c.tick(fps), second_len/fps, delta):
+            self._tick_test(reccnt + 1)
+        
+        # incur delay between ticks that's faster than fps 
+        pygame.time.wait(10)  
+        if not _is_almost_equal(c.tick(fps), second_len/fps, delta):
+            self._tick_test(reccnt + 1)
+        
+        # incur delay between ticks that's slower than fps
+        pygame.time.wait(200)  
+        # the function must return a value close to 200
+        if not _is_almost_equal(c.tick(fps), 200, delta):
+            self._tick_test(reccnt + 1)
+        
+        # Test a wide range of FPS with all kinds of values
+        for fps in [500, 1, 35, 600, 32.75]:
+            c = Clock()
+            if not _is_almost_equal(c.tick(fps), second_len/fps, delta):
+                self._tick_test(reccnt + 1)
+                
         zero_fps = 0
-        self.assertEqual(c.tick_busy_loop(zero_fps), 0)
-
-        # Check behaviour of unexpected values
-
+        self.assertAlmostEqual(c.tick(zero_fps), 0, delta=0.5)
         negative_fps = -1
-        self.assertEqual(c.tick_busy_loop(negative_fps), 0)
-
-        fractional_fps = 32.75
-        frame_length_without_decimal_places_3 = int(second_length/fractional_fps)
-        self.assertGreaterEqual(c.tick_busy_loop(fractional_fps), frame_length_without_decimal_places_3 - shortfall_tolerance)
-
-        bool_fps = True
-        self.assertGreaterEqual(c.tick_busy_loop(bool_fps), (second_length/bool_fps) - shortfall_tolerance)
-
-
+        self.assertAlmostEqual(c.tick(negative_fps), 0, delta=0.5)
+            
 class TimeModuleTest(unittest.TestCase):
```

Comment:
Should be an extra line break before the start of the class definition.

Reply:
Done :)